products/dialogflow/api.yaml

```diff
@@ -0,0 +1,123 @@
+        description: |
+         The list of all languages supported by this agent (except for the defaultLanguageCode).
+      - !ruby/object:Api::Type::String
+        name: 'timeZone'
+        description: |
+         The time zone of this agent from the [time zone database](https://www.iana.org/time-zones), e.g., America/New_York, Europe/Paris.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+         The description of this agent. The maximum length is 500 characters. If exceeded, the request is rejected.
+      - !ruby/object:Api::Type::String
+        name: 'avatarUri'
+        description: |
+         The URI of the agent's avatar, which are used throughout the Dialogflow console. When an image URL is entered 
+         into this field, the Dialogflow will save the image in the backend. The address of the backend image returned 
+         from the API will be shown in the [avatarUriBackend] field.
+      - !ruby/object:Api::Type::String
+        name: 'avatarUriBackend'
+        description: |
+         The URI of the agent's avatar as returned from the API. Output only. To provide an image URL for the agent avatar,
+         the [avatarUri] field can be used.
+        output: true
+      - !ruby/object:Api::Type::Boolean
+        name: 'enableLogging'
+        description: |
+         Determines whether this agent should log conversation queries.
+      - !ruby/object:Api::Type::Enum
+        name: 'matchMode'
+        description: |
+          Determines how intents are detected from user queries.
+          * MATCH_MODE_HYBRID: Best for agents with a small number of examples in intents and/or wide use of templates syntax and composite entities.
+          * MATCH_MODE_ML_ONLY: Can be used for agents with a large number of examples in intents, especially the ones using @sys.any or very large developer entities.
+        values:
+          - :MATCH_MODE_HYBRID
+          - :MATCH_MODE_ML_ONLY
+      - !ruby/object:Api::Type::Double
+        name: 'classificationThreshold'
+        description: |
+         To filter out false positive results and still get variety in matched natural language inputs for your agent, you can tune the machine learning classification threshold. 
+         If the returned score value is less than the threshold value, then a fallback intent will be triggered or, if there are no fallback intents defined, no intent will be 
+         triggered. The score values range from 0.0 (completely uncertain) to 1.0 (completely certain). If set to 0.0, the default of 0.3 is used.
+      - !ruby/object:Api::Type::Enum
+        name: 'apiVersion'
+        description: |
+          API version displayed in Dialogflow console. If not specified, V2 API is assumed. Clients are free to query different service endpoints for different API versions. 
+          However, bots connectors and webhook calls will follow the specified API version.
```

Comment:
It sounds like you can call a beta url with a `v2` version and a GA url with `v2_beta` is this correct? Also if you are defaulting to `v2` here it will be the default for both the GA and Beta provider - is this your intent?

Reply:
That is correct, it is my understanding this is more for downstream resources and enabling beta features. 
The API itself defaults to `V2`, independent of whether it is called from the GA or beta url